common/namer/namer.go

```diff
@@ -1,57 +1,131 @@
 package namer
 
 import (
+	"fmt"
+	"hash/fnv"
+	"io"
+	"math"
 	"strings"
 
 	"github.com/pulumi/pulumi/sdk/v3/go/pulumi"
+	"github.com/samber/lo"
 )
 
 const nameSep = "-"
 
+// In case of truncation, size of the hash suffix in percentage of the full size
+const hashSizePercent = 10
```

Comment:
<!-- Thoughts represent an idea that popped up from reviewing. These comments are non-blocking by nature, but they are extremely valuable and can lead to more focused initiatives and mentoring opportunities. -->
**💭 thought**
I can see this being helpful for short names, should we also have an upper bound with `maxHashSize` ? More for readability than for it being used at anytime, we don't expect to cut names for resources limited to 400 chars and thus using 40 chars of hash

Reply:
In fact, there’s already an implicit upper bound to the size of the hash suffix that comes from the hash function.
We’re using a hash function that returns a 64 bits long hash. So, it’s hexadecimal representation is 16 characters long.
In the case we have a field limited to 400 chars and so many long tokens that a truncation is needed, the hash suffix will be exactly 16 char long.
So, introducing yet another `maxHashSize` parameter might be overkill. WDYT ?
But I can add a test with a 400 char limit to validate this behavior and make it more explicit for code readers ?

Reply:
Ah good, I thought you would have used 40 chars of hash on a 400 chars name. Ok if it is already upper limited to the max hash size !